src/mpyl/steps/run_properties.py

```diff
@@ -16,53 +15,42 @@ def construct_run_properties(
     config: dict,
     properties: dict,
     cli_parameters: MpylCliParameters = MpylCliParameters(),
-    run_plan: Optional[RunPlan] = None,
-    all_projects: Optional[set[Project]] = None,
     root_dir: Path = Path(""),
     explain_run_plan: bool = False,
 ) -> RunProperties:
-    tag = cli_parameters.tag or properties["build"]["versioning"].get("tag")
-    if all_projects is None or run_plan is None:
-        with Repository(RepoConfig.from_config(config)) as repo:
-            if all_projects is None:
-                project_paths = repo.find_projects()
-                all_projects = set(
-                    map(
-                        lambda p: load_project(
-                            root_dir=root_dir,
-                            project_path=Path(p),
-                            strict=False,
-                            log=True,
-                        ),
-                        project_paths,
-                    )
-                )
+    with Repository(RepoConfig.from_config(config)) as repo:
+        all_projects = set(
+            map(
+                lambda p: load_project(
+                    root_dir=root_dir,
+                    project_path=Path(p),
+                    strict=False,
+                    log=True,
+                ),
+                repo.find_projects(),
+            )
+        )
 
-            if run_plan is None:
-                stages = [
-                    Stage(stage["name"], stage["icon"])
-                    for stage in properties["stages"]
-                ]
-                run_plan_logger = logging.getLogger("mpyl")
-                if explain_run_plan:
-                    run_plan_logger.setLevel("DEBUG")
-                changed_files_path = config["vcs"].get("changedFilesPath", None)
-                run_plan = _create_run_plan(
-                    cli_parameters=cli_parameters,
-                    all_projects=all_projects,
-                    all_stages=stages,
-                    explain_run_plan=explain_run_plan,
-                    repo=repo,
-                    tag=tag,
-                    changed_files_path=changed_files_path,
-                )
+        stages = [Stage(stage["name"], stage["icon"]) for stage in properties["stages"]]
+        run_plan_logger = logging.getLogger("mpyl")
+        if explain_run_plan:
+            run_plan_logger.setLevel("DEBUG")
+        changed_files_path = config["vcs"].get("changedFilesPath", None)
```

Comment:
and otherwise (if you don't want to change the schema) I think it's better to check for it right after loading it here

Reply:
That's exactly what I wanted to do, but unfortunately we can't because we only check for an existing run plan in disk 10 stack calls later, meaning we only know if we need this file or not much later in the process 😢

I'll refactor this in a follow-up PR.